packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -310,16 +316,74 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
       .filter((item): item is Target => !!item);
   }, [targetInfoPrompts, targetData]);
 
+  const EMAIL_REGEX = /^[a-zA-Z0-9._:$!%-+]+@[a-zA-Z0-9.-]+.[a-zA-Z]$/;
+  const TELEGRAM_REGEX =
+    /^@?(?=\w{5,32}\b)[a-zA-Z0-9]+(?:[a-zA-Z0-9_ ]+[a-zA-Z0-9])*$/;
+
+  const validateEmail = (
+    email: string,
+  ): { isValid: boolean; error?: string } => {
+    if (email === '') {
+      return { isValid: true };
+    }
+    const isValid = EMAIL_REGEX.test(email);
+    return {
+      isValid,
+      error: isValid ? undefined : 'Invalid email address',
+    };
+  };
+
```

Comment:
In context, we might want to simply just give the error message if there is. 
And in component, we can decide the component logic more flexible using the util method mentioned ablve